src/components/Select/Select.tsx

```diff
@@ -48,7 +49,7 @@ function Select<T extends object>(props: Props<T>, ref: RefObject<HTMLDivElement
 
   const state = useSelectState(props);
   const { labelProps, triggerProps, valueProps, menuProps } = useSelect(props, state, selectRef);
-  const { buttonProps } = useButton({ ...triggerProps, isDisabled }, selectRef);
+  const { buttonProps } = useButton({ ...triggerProps, isDisabled, ...(ariaLabelledBy === '' ? {'aria-labelledby': null} : ariaLabelledBy ? { 'aria-labelledby': ariaLabelledBy } : {})}, selectRef);
```

Comment:
non-blocking: Not a fan of nested ternaries, and this is a bit hard to read.
```suggestion
  const { buttonProps } = useButton(
    { ...triggerProps, isDisabled, 'aria-labelledby': ariaLabelledBy || null },
    selectRef
  );
 ```
 
 Is this close enough?  It looks like you specifically wanted to add nothing if `ariaLabelledBy` is nullish but does it really matter?